tool/tctl/common/auth_command.go

```diff
@@ -671,6 +671,19 @@ func (a *AuthCommand) generateUserKeys(ctx context.Context, clusterAPI auth.Clie
 		kubeTLSServerName = client.GetKubeTLSServerName(a.config.Proxy.WebAddr.Host())
 	}
 
+	expires, err := key.TeleportTLSCertValidBefore()
+	if err != nil {
+		log.WithError(err).Warn("Failed to check TTL validity")
+		// err swallowed on purpose
+		return nil
+	}
+	if reqExpiry.Sub(expires) > time.Minute {
+		maxAllowedTTL := time.Until(expires).Round(time.Second)
+		return trace.Errorf(`The credential was not issued because the requested TTL of %s exceeded the maximum allowable value of %s. To successfully request a credential, please reduce the requested TTL.`,
```

Comment:
nit: I've been told to avoid naked trace.Errorf.

```suggestion
		return trace.BadParameter(`The credential was not issued because the requested TTL of %s exceeded the maximum allowed value of %s. To successfully request a credential, please reduce the requested TTL.`,
```